File: qb-engineer-server/qb-engineer.core/Models/CreateQuoteRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreateQuoteRequestModel(
    int CustomerId,
    int? ShippingAddressId,
    DateTime? ExpirationDate,
    string? Notes,
    decimal TaxRate,
    List<CreateQuoteLineModel> Lines);

public record CreateQuoteLineModel(
    int? PartId,
    string Description,
    int Quantity,
    decimal UnitPrice,
    string? Notes);

File: qb-engineer-ui/src/app/features/quotes/models/create-quote-request.model.ts
import { CreateQuoteLineRequest } from './create-quote-line-request.model';

export interface CreateQuoteRequest {
  customerId: number;
  shippingAddressId?: number;
  expirationDate?: string;
  notes?: string;
  taxRate: number;
  lines: CreateQuoteLineRequest[];
}
